test/spmd/test_xla_sharding.py

```diff
@@ -95,6 +95,20 @@ def test_replicated_xla_shards(self):
       self.assertTrue(torch.allclose(shard.data, t[shard.indices]))
       self.assertTrue(torch.allclose(shard.data, shard.unpadded_data))
 
+  def test_load_local_shards(self):
```

Comment:
Does this test the replicated path?

Reply:
The `load_local_shards` doesn't support replicated tensors - there would be a lot of overhead ensuring the input shards are actually the same.

Reply:
I know that, and you have added a bunch of XLA_CHECK. Should we test the exception as well?

Reply:
Thanks Jiewen! I didn't realize we could `assertRaises` on XLA_CHECK failures - I've added some more tests to cover the failure modes